### PR TITLE
Rename and expand timer bar util packet

### PR DESCRIPTION
**Pull request by cocosolos**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Renames timer bar util packet (incoming 0x075) to better reflect functionality.

- Add adjustable warning to timer function.
- Add support for up to 5 progress bars.
- Add support for adjustable battle fence.
- Add limited support for help text display.
- Remove unused duplicate Lua binding.
- Adjust existing countdown binding and usage.

There's still a few unknowns that I've left commented, and I made a few guesses based on some captures of different uses of this packet, including: UNM, Domain Invasion/Geas Fete, Dynamis D, Odyssey, HTMB, and the Mandragora Mania mini-game.

## Steps to test these changes

Add the following command to `scripts\commands\objective.lua` and use `!objective` in game on a GM character. You can adjust the examples to test various aspects. Documentation in lua_baseentity.cpp.

```lua
local commandObj = {}

commandObj.cmdprops =
{
    permission = 1,
    parameters = ''
}

local function error(player, msg)
    player:printToPlayer(msg)
    player:printToPlayer('!objective')
end

commandObj.onTrigger = function(player)
    local objective = {
        countdown = 65,
        --[[countdown = {
            duration = 60,
            warning = 30,
        },]]
        bars = {
            [1] = {
                title = "", -- hides bar
                value = 0
            },
            [2] = {
                title = " ",
                value = 25
            },
            [3] = {
                title = "^25% v50%",
                value = 50
            },
            -- 4 is skipped
            [5] = {
                title = ">100%",
                value = 101
            },
        },
        fence = {
            pos = {x = 0, y = 0},
            radius = 25,
            render = 25
        },
        help = {
            title = 12,
            description = 52
        }
    }

    player:objectiveUtility(objective)
end

return commandObj
```

To test Full Speed Ahead, use the following commands in game on a GM character and talk to Mapitoto.
```
!completequest 3 4
!addkeyitem 388
!addkeyitem 138
!gotoid 17776895
```

### CODE REVIEW

**Comment by WinterSolstice8** [comment]
Not sure there's a reason to rename it when it's clearly used for more than just unity

**Comment by cocosolos** [comment]
TBF it's used for more than a timer as well. I'm not thrilled with Unity, but I assume it started being used around that time and is what it's being called by others.

File: src/map/packets/unity_start.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+#include "unity_start.h"
+
+#include "entities/charentity.h"
+
+CUnityStartPacket::CUnityStartPacket()
+{
+    this->setType(0x75);
+    this->setSize(0xAC);
+
+    // ref<uint8>(0x25) = ; // Flags value with multiple usages.
```
Comment (zach2good) [inline src/map/packets/unity_start.cpp]: Any chance you can expand on these addresses you've commented out with any notes and findings you might have, even if they aren't implemented this time around?

**Review by zach2good — COMMENTED** [review]
Great work, cool to see this packet getting fleshed out!

I'm also not so keen on it being called UnityStart; given that it was in use long before Unity fights were a thing - other sources calling it X doesn't stop us calling it Y if we want to.

I quite like you passing in a table called 'objective', that might be a good root for a name

**Comment by cocosolos** [comment]
> given that it was in use long before Unity fights were a thing

Was it? It was my understanding that the timer in BCNMs was a relatively recent addition, and the furthest I can trace it back is either the 2013 or 2014 anniversary event based on the help text strings, which would coincide with the release of Unity in 2014. Of the content that uses it that I know of UNM, Domain Invasion, and Odyssey are all Unity adjacent (DI having the call to action in Unity chat, and Odyssey giving UNM upgrade materials). That leaves just Dynamis D, Mandragora Mania, and Full Speed Ahead to my knowledge, other than the BCNM timer, and those are all more recent than Unity.

I originally was calling it ObjectiveUtility if you prefer that. I just figured it would make sense to have a unified (heh) name within the community, but I'm open to whatever. Just thought "timer bar" was limited; consider this use for DI:
```lua
player:unityStart({fence = {radius = 25}})
player:objectiveUtility({fence = {radius = 25}})
player:timerBarUtil({fence = {radius = 25}})
```
Since DI doesn't use the timer or progress bar functions, which one looks best?

File: src/map/packets/unity_start.cpp (hunk 1 above)
Comment (cocosolos) [reply src/map/packets/unity_start.cpp]: I actually haven't figured out or done much testing on what these 2 bytes do. The comments about them being flag values are courtesy of Atom0s.

Comment (cocosolos) [reply src/map/packets/unity_start.cpp]: The only thing I really noticed about them is that 0x25 changes frequently in Dynamis.

**Comment by zach2good** [comment]
My assumption was the 'Full Speed Ahead' would have come in before Unity, but it looks like that came in in 2016 🤷 

For me `objectiveUtility` looks best. The packet might have come in with Unity stuff, but it's used in many other things, so a more wider-reaching name is better. It'll be clear how/when it's used when there are some examples. I'm sure that now the radius stuff is figured out someone is going to implement a single Unity battle

File: src/map/packets/unity_start.cpp (hunk 1 above)
Comment (zach2good) [reply src/map/packets/unity_start.cpp]: If it's usable for unity/DI/current stuff as-is, then don't worry about trying to find more notes

**Review by zach2good — APPROVED** [review]

**Comment by zach2good** [comment]
Now WTB someone to implement (properly) everything that goes into a single Unity battle to set a pattern for everyone else to use (I know everyone else never appears, but I'll keep on hoping).

**Pull request merged**